src/types/markers.js

```diff
@@ -7,6 +7,49 @@ import type { Milliseconds, Microseconds, Seconds, Bytes } from './units';
 import type { GeckoMarkerStack } from './gecko-profile';
 import type { IndexIntoStackTable, IndexIntoStringTable } from './profile';
 
+// Provide different formatting options for strings.
+export type MarkerFormatType =
+  | 'string' // "Label: Some String"
+  | 'bytes' // "Label: 5mb"
+  | 'seconds' // "Label: 5s"
+  | 'milliseconds' // "Label: 5ms"
+  | 'microseconds' // "Label: 5μs"
+  | 'integer' // "Label: 5323"
+  | 'number' // "Label: 52.23"
+  | 'percentage' // "Label: 50%"
+  | 'stack' // Prints a full stack trace using the value.
+  | 'url'; // Show the URL, and handle sanitization
+
+// A list of all the valid locations to surface this marker.
+// We can be free to add more UI areas.
+export type MarkerDisplayLocations =
```

Comment:
nit: I'd remove the final `s`. When we'll use it, this is only ever holding one `MarkerDisplayLocation`.

Reply:
Agreed, this was awkward in later patches. I'll handle it now.